Fix part of #13956: Remove python_utils.RANGE variable

megamayoy opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #13956.
2. This PR does the following: remove the compatibility code for **range()** which were written to support Python 2 and replace it with its corresponding **range()** function in Python 3 

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
Pre-submit checks pass succesfully.
Since all my changes are in the back-end, I made sure that all backend tests pass successfully.
## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

Commit:
replace python_utils.RANGE with Python 3's range function

After migrating to Python 3, There're some variables in python_utils.py that are no longer
needed, so python_utils.RANGE variable is replaced with Python's range function.

Commit:
Remove python_utils.RANGE variable

After replacing all usages of python_utils.RANGE with Python's range functions,
the python_utils.RANGE variable should be removed to clean up python_utils.py file

Commit:
remove unused python_utils imported from core

after deleting python_utils.RANGE variable from the entire codebase, unused imports of python_utils appeared, so they should be deleted.

Commit:
Remove python_utils.RANGE from disallowed-functions in .pylintrc

python_utils.RANGE is removed from .pylintrc as it longer exists

DubeySandeep commented:
@megamayoy Can you please follow the checklist and update the PR title?

megamayoy commented:
> @megamayoy Can you please follow the checklist and update the PR title?

@DubeySandeep  Done

vojtechjelinek commented on core/controllers/learner_goals_test.py:
```diff
@@ -159,7 +159,7 @@ def test_add_topic_to_learner_goal(self):
         # the maximum limit.
-        for topic_id in python_utils.RANGE(
+        for topic_id in range(
                 2, feconf.MAX_CURRENT_GOALS_COUNT + 1):
```
Move this to the previous line.

megamayoy replied:
Done

vojtechjelinek commented on core/controllers/learner_playlist_test.py:
```diff
@@ -148,7 +148,7 @@ def test_add_exploration_to_learner_playlist(self):
         # the maximum limit.
-        for exp_id in python_utils.RANGE(
+        for exp_id in range(
                 5, feconf.MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT + 3):
```
Move this to the previous line.

megamayoy replied:
Done

vojtechjelinek commented on core/controllers/questions_list_test.py:
```diff
@@ -66,7 +65,7 @@ def setUp(self):
 
     def test_get_questions_succeeds(self):
-        for _ in python_utils.RANGE(0, 4):
+        for _ in range(0, 4):
```
```suggestion
        for _ in range(4):
```

megamayoy replied:
Done

vojtechjelinek commented on core/controllers/questions_list_test.py:
```diff
@@ -96,7 +95,7 @@ def test_get_questions_succeeds(self):
                 json_response['question_summary_dicts'])
             self.assertEqual(len(question_summary_dicts_2), 2)
-            for i in python_utils.RANGE(0, 2):
+            for i in range(0, 2):
```
```suggestion
            for i in range(2):
```

megamayoy replied:
Done

vojtechjelinek commented on core/controllers/questions_list_test.py:
```diff
@@ -122,7 +121,7 @@ def test_get_questions_succeeds(self):
                 json_response['question_summary_dicts'])
             self.assertEqual(len(question_summary_dicts_3), 2)
-            for i in python_utils.RANGE(0, 2):
+            for i in range(0, 2):
```
```suggestion
            for i in range(2):
```

vojtechjelinek commented on core/controllers/reader_test.py:
```diff
@@ -343,7 +342,7 @@ def test_filter_multiple_skill_id_return_questions(self):
         # Create multiple skills, questions and skill links.
-        for _ in python_utils.RANGE(
+        for _ in range(
                 feconf.MAX_QUESTIONS_FETCHABLE_AT_ONE_TIME):
```
Move this to the previous line.

vojtechjelinek commented on core/domain/app_feedback_report_domain_test.py:
```diff
@@ -1199,7 +1199,7 @@ def test_validation_ticket_name_is_not_a_string_fails(self) -> None:
         long_name = 'too long'
-        for _ in python_utils.RANGE(constants.MAXIMUM_TICKET_NAME_LENGTH):
+        for _ in range(constants.MAXIMUM_TICKET_NAME_LENGTH):
             long_name += 'x'
```
```python
        long_name = 'too long' + 'x' * constants.MAXIMUM_TICKET_NAME_LENGTH
```

vojtechjelinek commented on core/domain/collection_domain.py:
```diff
@@ -783,7 +783,7 @@ def get_next_exploration_id_in_sequence(self, current_exploration_id):
         exploration_just_unlocked = None
 
-        for index in python_utils.RANGE(0, len(self.nodes) - 1):
+        for index in range(0, len(self.nodes) - 1):
```
```suggestion
        for index in range(len(self.nodes) - 1):
```

vojtechjelinek commented on core/domain/collection_domain_test.py:
```diff
@@ -716,7 +715,7 @@ def test_correct_collection_contents_schema_conversion_methods_exist(self):
             feconf.CURRENT_COLLECTION_SCHEMA_VERSION)
-        for version_num in python_utils.RANGE(
+        for version_num in range(
                 1, current_collection_schema_version):
```
Move this to the previous line.

vojtechjelinek commented on core/domain/collection_domain_test.py:
```diff
@@ -734,7 +733,7 @@ def test_correct_collection_schema_conversion_methods_exist(self):
 
-        for version_num in python_utils.RANGE(
+        for version_num in range(
                 1, current_collection_schema_version):
```
Move this to the previous line.

vojtechjelinek commented on core/domain/draft_upgrade_services.py:
```diff
@@ -73,7 +73,7 @@ def try_upgrading_draft_to_exp_version(
 
     exp_versions = list(
-        python_utils.RANGE(current_draft_version + 1, to_exp_version + 1))
+        range(current_draft_version + 1, to_exp_version + 1))
```
Move this to the previous line.

vojtechjelinek commented on core/domain/feedback_domain.py:
```diff
@@ -121,7 +121,7 @@ def get_last_two_message_ids(self):
             self._get_full_message_id(i) if i >= 0 else None
-            for i in python_utils.RANGE(
+            for i in range(
                 self.message_count - 1, self.message_count - 3, -1)
```
Move this to the previous line.

vojtechjelinek commented on core/domain/feedback_services.py:
```diff
@@ -728,7 +728,7 @@ def get_exp_thread_summaries(user_id, thread_ids):
         flattened_last_two_message_models_of_threads[i:i + 2]
-        for i in python_utils.RANGE(
+        for i in range(
             0, len(flattened_last_two_message_models_of_threads), 2)
```
Move this to the previous line.

vojtechjelinek commented on core/domain/improvements_services_test.py:
```diff
@@ -172,7 +171,7 @@ def test_fetch_when_no_models_exist(self):
             self._new_open_task(state_name='State %d' % (i,))
-            for i in python_utils.RANGE(
+            for i in range(
                 int(feconf.MAX_TASK_MODELS_PER_FETCH * 2.5))
```
Move this to the previous line.

vojtechjelinek commented on core/domain/suggestion_services.py:
```diff
@@ -877,7 +877,7 @@ def get_suggestions_waiting_for_review_info_to_notify_reviewers(reviewer_ids):
         reviewer_reviewable_suggestion_infos = []
-        for _ in python_utils.RANGE(
+        for _ in range(
                 MAX_NUMBER_OF_SUGGESTIONS_TO_EMAIL_REVIEWER):
```
Move this to the previous line.

vojtechjelinek commented on core/domain/suggestion_services_test.py:
```diff
@@ -2686,7 +2686,7 @@ def _assert_reviewable_suggestion_email_infos_are_in_correct_order(
                     index].submission_datetime)
-        for index in python_utils.RANGE(
+        for index in range(
                 len(reviewable_suggestion_email_infos) - 1):
```
Move this to the previous line.

vojtechjelinek commented on core/domain/suggestion_services_test.py:
```diff
@@ -3797,7 +3797,7 @@ def _assert_reviewable_suggestion_email_infos_are_in_correct_order(
                     index].submission_datetime)
-        for index in python_utils.RANGE(
+        for index in range(
                 len(reviewable_suggestion_email_infos) - 1):
```
Move this to the previous line.

megamayoy commented:
@vojtechjelinek Regarding the failing end-to-end test, the logs shows the message 

> E2E logging server says test flaky: True.. 

Are we gonna restart the tests until the failing tests passes successfully?

vojtechjelinek reviewed (changes requested):
Thanks, left a few comments.

vojtechjelinek commented:
> @vojtechjelinek Regarding the failing end-to-end test, the logs shows the message
> 
> > E2E logging server says test flaky: True..
> 
> Are we gonna restart the tests until the failing tests passes successfully?

Yes

Commit:
Delete some line breaks

This is done to make the new changes compliant with the general coding style

megamayoy commented:
@vojtechjelinek  All requested changes are done.

vojtechjelinek commented on core/controllers/learner_playlist_test.py:
```diff
@@ -247,7 +246,7 @@ def test_add_collection_to_learner_playlist(self):
         # the maximum limit.
-        for exp_id in python_utils.RANGE(
+        for exp_id in range(
                 5, feconf.MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT + 3):
```
Move this to the previous line.

megamayoy replied:
Done

vojtechjelinek reviewed (changes requested):
One last comment that I forgot to add before.

vojtechjelinek commented:
@seanlip Do you think this routine change needs to go through all the code owners?

Commit:
Remove a line break

remove a line break to make changes compatible with the general code style
of the repository

seanlip commented:
@vojtechjelinek No, I don't think so. Once you approve it, feel free to assign it to me and I can review/merge it. Thanks!

vojtechjelinek reviewed (approved):
LGTM!

vojtechjelinek commented:
@seanlip Ready to be merged.

Hudda reviewed (approved):
LGTM for codeowner files.

seanlip commented:
Merging. Congrats @megamayoy on your first PR!
